### PR TITLE
Add docs about connection details

### PR DESCRIPTION
**Pull request by Kidswiss**
## Summary

* Detailed documentation how to handle connection details with the composition function framework

## Checklist

- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog
- [x] Update tests.
- [x] Link this PR to related issues.

### CODE REVIEW

File: docs/modules/ROOT/pages/explanations/connectiondetails.adoc [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+	}
+----
+<1> The connection detail still needs to written somewhere
+<2> We don't have to bother with base64 encoding, provider-helm will handle this for us
+
+With this method it should be easier to handle most cases.
+However, provider-helm's connection detail management isn't perfect.
+It cannot handle any transformations on the secrets.
+So if anything needs to concatenated, split, or changed in any other way, then this might unfortunately not be the way to do it.
+
```
Comment (TheBigLee) [inline docs/modules/ROOT/pages/explanations/connectiondetails.adoc:156]: It should probably also be documented that you need to add that connection details to the composite as pointed in the above example with `return iof.Desired.PutWithResourceName(ctx, user, "minio-user", runtime.AddDerivedConnectionDetails(cd))`

File: docs/modules/ROOT/pages/explanations/connectiondetails.adoc [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+if err != nil {
+  if err == runtime.ErrNotFound { <1>
+    return runtime.NewNormal()
+  }
+	return runtime.NewFatalErr(ctx, "cannot get observed deletion job", err)
+}
+----
+<1> We need to ignore not found errors here, as the observer will not exist during the first reconcile
+
+Also, here for secrets the values will not be base64 encoded, so they can be used directly.
```
Comment (TheBigLee) [inline docs/modules/ROOT/pages/explanations/connectiondetails.adoc:196]: I also expect some kind of documentation on how to add the gathered data to the composite here. Eg:
```
iof.Desired.PutCompositeConnectionDetail(ctx, v1alpha1.ExplicitConnectionDetail{
	Name:  "MY_PASSWORD",
	Value: string(secret.Data["my_password"])
})
```

File: docs/modules/ROOT/pages/explanations/connectiondetails.adoc [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+			WriteConnectionSecretToReference: &xpv1.SecretReference{
+				Name:      "uniqname", <1>
+				Namespace: "secretnamespace", <2>
+			},
+		},
+	},
+}
+----
+<1> Unique name for this secret, usually the composite name + some suffix
+<2> The namespace where the secret should be saved, usually set to `writeConnectionSecretsToNamespace`.
```
Comment (zugao) [inline docs/modules/ROOT/pages/explanations/connectiondetails.adoc:56]: `writeConnectionSecretsToNamespace` - we can specify that this is from composition.

**Review by zugao — APPROVED** [review]
Looks good. thanks for this valuable doc. There's just one concern and one typo.

**Pull request merged**